internal/controller/controllers/cluster_controller.go

```diff
@@ -233,6 +237,14 @@ func (r *ClusterReconciler) updateIfNeeded(ctx context.Context, cluster *adiiov1
 		}
 	}
 
+	updatePullSecret := func(new, old string, target **string) {
```

Comment:
i don't think that you need to create this function, this logic is called once, it will be clearer to have the code in a right place.

Reply:
Ack. will change that.